### PR TITLE
enhancement: use lock_guard to replace manual.

### PR DESCRIPTION
**Pull request by jackwener**

### CODE REVIEW

File: src/worker.cc [hunk 1]
```diff
@@ -345,10 +345,9 @@ Status Worker::Reply(int fd, const std::string &reply) {
 }
 
 void Worker::BecomeMonitorConn(Redis::Connection *conn) {
-  conns_mu_.lock();
+  std::lock_guard<std::mutex> guard(conns_mu_);
```
Comment (PragmaTwice) [inline src/worker.cc]: sematics is not equal to the previous code here

File: src/worker.cc [hunk 2]
```diff
@@ -421,7 +418,6 @@ void Worker::KickoutIdleClients(int timeout) {
   }
   iter--;
   last_iter_conn_fd = iter->first;
-  conns_mu_.unlock();
```
Comment (PragmaTwice) [inline src/worker.cc:424]: same as above

**Comment by jackwener** [comment]
Thanks! 
Has fixed it.

**Review by git-hulk — APPROVED** [review]
LGTM

**Review by PragmaTwice — APPROVED** [review]

**Comment by git-hulk** [comment]
Thanks all, merging PR...

**Pull request merged**